src/components/NavigationHeader.tsx

```diff
@@ -0,0 +1,45 @@
+'use client';
+
+import { useNavigation } from '@/contexts/NavigationProvider';
+import { Box, Typography } from '@mui/material';
+import Image from 'next/image';
+import { useRouter } from 'next/navigation';
+
+export default function NavigationHeader() {
+  const { title, headerColor } = useNavigation();
+  const router = useRouter();
+
+  return (
+    <Box
+      onClick={() => router.back()}
```

Comment:
As we know all the paths of the site, it is preferable to create the return url using pathname